searchlib/src/vespa/searchlib/attribute/elementiterator.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "elementiterator.h"
+#include <vespa/searchcommon/attribute/i_search_context.h>
+#include <vespa/searchlib/fef/termfieldmatchdata.h>
+
+using search::fef::TermFieldMatchDataPosition;
+
+namespace search::attribute {
+
+void
+ElementIterator::doSeek(uint32_t docid) {
+    _search->doSeek(docid);
+    setDocId(_search->getDocId());
+}
+
+void
+ElementIterator::doUnpack(uint32_t docid) {
+    _tfmda.reset(docid);
+    int32_t weight(0);
+    for (int32_t id = _searchContext.find(docid, 0, weight); id >= 0; id = _searchContext.find(docid, id+1, weight)) {
+        _tfmda.appendPosition(TermFieldMatchDataPosition(id, 0, weight, 1));
+    }
+}
+
+vespalib::Trinary
+ElementIterator::is_strict() const {
+    return _search->is_strict();
+}
+
+void
+ElementIterator::initRange(uint32_t beginid, uint32_t endid) {
+    _search->initRange(beginid, endid);
+    SearchIterator::initRange(_search->getDocId()+1, _search->getEndId());
```

Comment:
This is confusing and depends on implementation details. I suggest:
SearchIterator::initRange(beginId, endId);
_child->initRange(beginId, endId);
setDocId(_child->getDocId());

Reply:
Ok, this was copied from another wrapping iterator.